store/tikv/coprocessor.go

```diff
@@ -142,19 +167,117 @@ type copIterator struct {
 	store *tikvStore
 	req   *kv.Request
 	tasks []*copTask
-	index int // index indicates the next task to be executed
+
+	mu          sync.RWMutex
+	respGot     int
+	concurrency int
+	respChan    chan *coprocessor.Response
+	errChan     chan error
+	finished    bool
 }
 
-func (it *copIterator) Next() (io.ReadCloser, error) {
-	// TODO: Support `Concurrent` instruction in `kv.Request`.
+// Pick the next new copTask and send request to tikv-server.
+func (it *copIterator) work() {
+	for {
+		if it.finished {
+			break
+		}
+		it.mu.Lock()
+		// Find the next task to send.
+		var task *copTask
+		if it.req.Desc {
+			for i := len(it.tasks) - 1; i >= 0; i-- {
+				if it.tasks[i].status == taskNew {
+					task = it.tasks[i]
+					break
+				}
+			}
+		} else {
+			for _, t := range it.tasks {
+				if t.status == taskNew {
+					task = t
+					break
+				}
+			}
+		}
+		it.mu.Unlock()
+		if task == nil {
+			break
+		}
+		task.status = taskRunning
```

Comment:
I think you should update the status with the mutex acquired, or 2 worker may send the same task.